draft-schlesinger-cfrg-act.md

```diff
@@ -362,20 +349,40 @@ IssueRequest():
     1. k <- Zq  // Nullifier (will prevent double-spending)
     2. r <- Zq  // Blinding factor
     3. K = H2 * k + H3 * r
-    4. // Generate proof of knowledge of k, r
-    5. k' <- Zq
-    6. r' <- Zq
-    7. K1 = H2 * k' + H3 * r'
-    8. transcript = CreateTranscript("request")
-    9. AddToTranscript(transcript, K)
-    10. AddToTranscript(transcript, K1)
-    11. gamma = GetChallenge(transcript)
-    12. k_bar = k' + gamma * k
-    13. r_bar = r' + gamma * r
-    14. request = (K, gamma, k_bar, r_bar)
-    15. state = (k, r)
-    16. return (request, state)
+
+    // Generate proof of knowledge of (k, r) such that K = H2 * k + H3 * r
+    4. relation = build_relation_issue_request(H2, H3, K)
+    5. nizk = NISigmaProtocol("request", relation)
+    6. witness = [k, r]
+    7. pok = nizk.prove(witness)
+    8. request = (K, pok)
+    9. state = (k, r, K)
+   10. return (request, state)
+~~~
+
+The {{build_relation_issue_request}}{:format="title"} function
+constructs a linear relation to prove knowledge of the
+scalars `(k0, k1)` such that `R = k0*P + k1*Q`, for group
+elements `P`, `Q`, and `R`.
```

Comment:
```suggestion
The {{build_relation_issue_request}}{:format="title"} function
constructs a linear relation to prove knowledge of a Pederson commitment, that is, knowledge of
scalars `(k0, k1)` such that `R = k0*P + k1*Q`, for group
elements `P`, `Q`, and `R`.
```

Reply:
I might even call this function something like `build_relation_pederson_commitment()` and refer (informationally) to https://mmaker.github.io/draft-irtf-cfrg-sigma-protocols/draft-irtf-cfrg-sigma-protocols.html#section-2.2.9. The only difference between the code here and that section is the invocation of `set_elements()`.